hack/mkdocs/generate.sh

```diff
@@ -32,5 +32,13 @@ go install github.com/elastic/crd-ref-docs
 ${GOBIN}/crd-ref-docs \
     --source-path=${PWD}/apis \
     --config=crd-ref-docs.yaml \
+    --templates-dir=${PWD}/hack/crd-ref-templates/ \
     --renderer=markdown \
     --output-path=${PWD}/site-src/reference/spec.md
+
+${GOBIN}/crd-ref-docs \
```

Comment:
@youngnick @robscott I am not sure if we were intentionally not generating the experimental API docs, but I've figured out that the current version on our docs are outdated

https://gateway-api.sigs.k8s.io/reference/specx/

Even Mesh resources are not here so far, do we expect that experimental is manually generated?

Reply:
I think during some refactor around when we introduced the `x-k8s.io` API group for new experimental resources we inadvertently stopped rendering docs for older experimental resources that predated that change.

Reply:
Yeah, it's almost certainly just an oversight. We should be generating the experimental API docs, for sure.